### PR TITLE
Apply various contentual and stylish improvements to the k8s docs.

### PR DESCRIPTION
**Pull request by timoreimann**
This includes addressing a number of markdown linter violations.

Details:

- Improve terminology.
- Rephrase several sentences.
- Group service-based annotations consistently.
- Add note about Helm chart maintainership.
- Fix headline levels.
- Fix headline titles.
- Fix grammar.
- Improve formatting.
- Remove trailing white-spaces.
- Remove extra white-spaces.
- Specify language on fenced code blocks.
- Surround fenced clode blocks by blank line.

### CODE REVIEW

File: docs/configuration/backends/kubernetes.md [hunk 1]
```diff
@@ -75,30 +74,31 @@ See also [Kubernetes user guide](/user-guide/kubernetes).
...
 ### `endpoint`
 
-The Kubernetes server endpoint.
+The Kubernetes server endpoint as URL.
 
-When deployed as a replication controller in Kubernetes, Traefik will use the environment variables `KUBERNETES_SERVICE_HOST` and `KUBERNETES_SERVICE_PORT` to construct the endpoint.
+When deployed into Kubernetes, Traefik will read the environment variables `KUBERNETES_SERVICE_HOST` and `KUBERNETES_SERVICE_PORT` to construct the endpoint.
 
-Secure token will be found in `/var/run/secrets/kubernetes.io/serviceaccount/token` and SSL CA cert in `/var/run/secrets/kubernetes.io/serviceaccount/ca.crt`
+The access token will be looked up in `/var/run/secrets/kubernetes.io/serviceaccount/token` and the SSL CA certificate in `/var/run/secrets/kubernetes.io/serviceaccount/ca.crt`. Both are provided mounted automatically when deployed inside Kubernetes.
```
Comment (nmengin) [inline docs/configuration/backends/kubernetes.md]: Can you use one line per sentence?

File: docs/configuration/backends/kubernetes.md [hunk 2]
```diff
@@ -75,30 +74,31 @@ See also [Kubernetes user guide](/user-guide/kubernetes).
...
 
 When the environment variables are not found, Traefik will try to connect to the Kubernetes API server with an external-cluster client.
 In this case, the endpoint is required.
-Specifically, it may be set to the URL used by `kubectl proxy` to connect to a Kubernetes cluster from localhost.
+Specifically, it may be set to the URL used by `kubectl proxy` to connect to a Kubernetes cluster using the granted autentication and authorization of the associated kubeconfig.
 
 ### `labelselector`
 
-Ingress label selector to identify Ingress objects that should be processed.
+By default, Traefik processes all Ingress objects in the configured namespaces. A label selector can be defined to filter on specific Ingress objects only.
```
Comment (nmengin) [inline docs/configuration/backends/kubernetes.md]: Can you use one line per sentence?

File: docs/configuration/backends/kubernetes.md [hunk 3]
```diff
@@ -108,48 +108,39 @@ Annotations can be used on containers to override default behaviour for the whol
...
 - `traefik.frontend.redirect.replacement: http://mydomain/$1`:
     Redirect to another URL for that frontend. Must be set with `traefik.frontend.redirect.regex`.
-- `traefik.frontend.entryPoints: http,https`  
+- `traefik.frontend.entryPoints: http,https`
     Override the default frontend endpoints.
-- `traefik.frontend.passTLSCert: true`  
+- `traefik.frontend.passTLSCert: true`
     Override the default frontend PassTLSCert value. Default: `false`.
+- `ingress.kubernetes.io/whitelist-source-range: "1.2.3.0/24, fe80::/16"`
+    A comma-separated list of IP ranges permitted for access. all source IPs are permitted if the list is empty or a single range is ill-formated.
```
Comment (nmengin) [inline docs/configuration/backends/kubernetes.md]: Can you use one line per sentence and begin the second sentence by an upper-case?

I'm not sure but `ill-formated` shoud not be ` ill-formatted` ?

Comment (timoreimann) [reply docs/configuration/backends/kubernetes.md]: [You are right](https://en.wiktionary.org/wiki/formated). :-)

**Comment by timoreimann** [comment]
@nmengin addressed your comments.

Also rebased to fix a merge conflict and re-formatted an updated table.

File: docs/user-guide/kubernetes.md [hunk 4]
```diff
@@ -1,15 +1,17 @@
...
 
 If you are not familiar with Ingresses in Kubernetes you might want to read the [Kubernetes user guide](https://kubernetes.io/docs/concepts/services-networking/ingress/)
 
 The config files used in this guide can be found in the [examples directory](https://github.com/containous/traefik/tree/master/examples/k8s)
 
 ## Prerequisites
 
-1. A working Kubernetes cluster. If you want to follow along with this guide, you should setup [minikube](https://kubernetes.io/docs/getting-started-guides/minikube/)
-on your machine, as it is the quickest way to get a local Kubernetes cluster setup for experimentation and development.
+1. A working Kubernetes cluster. If you want to follow along with this guide, you should setup [minikube](https://kubernetes.io/docs/getting-started-guides/minikube/) on your machine, as it is the quickest way to get a local Kubernetes cluster setup for experimentation and development.
```
Comment (nmengin) [inline docs/user-guide/kubernetes.md]: I forgot to mention these sentences to split too.  :wink:

Comment (timoreimann) [reply docs/user-guide/kubernetes.md]: As this sentence is part of a list item, splitting it means that [MD032](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md032---lists-should-be-surrounded-by-blank-lines) is triggered. I'd opt for keeping it the way it is.

**Pull request merged**